pmacFilterControlWrapper.py

```diff
@@ -0,0 +1,366 @@
+        )
+        self.lower_high_threshold = builder.aOut(
+            "HIGH:THRESHOLD:LOWER",
+            initial_value=2,
+            on_update=self._set_lower_high_threshold,
+        )
+        self.upper_low_threshold = builder.aOut(
+            "LOW:THRESHOLD:UPPER",
+            initial_value=2,
+            on_update=self._set_upper_low_threshold,
+        )
+        self.lower_low_threshold = builder.aOut(
+            "LOW:THRESHOLD:LOWER",
+            initial_value=2,
+            on_update=self._set_lower_low_threshold,
+        )
+
+        self.filter_set = builder.mbbOut(
+            "FILTER_SET", *FILTER_SET, initial_value=0, on_update=self._set_filter_set
+        )
+        self.filter_set_rbv = builder.mbbIn(
+            "FILTER_SET_RBV", *FILTER_SET, initial_value=0
+        )
+
+        self.file_path = builder.stringOut("FILE:PATH", on_update=self._set_file_path)
+        self.file_name = builder.stringOut("FILE:NAME", on_update=self._set_file_name)
+        self.file_full_name = builder.stringIn("FILE:FULL_NAME")
+
+        self.process_duration = builder.aIn("PROCESS:DURATION")
+        self.process_period = builder.aIn("PROCESS:PERIOD")
+
+        self.last_frame_received = builder.aIn("FRAME:RECEIVED")
+        self.last_frame_processed = builder.aIn("FRAME:PROCESSED")
+        self.time_since_last_frame = builder.aIn("FRAME:LAST_TIME")
+
+        self.current_attenuation = builder.aIn("ATTENUATION_RBV")
+
+        self.filter_sets_in = {}
+        self.filter_sets_out = {}
+        for i in range(1, filter_set_total + 1):
+            filter_set_key = f"filter_set_{i}"
+            self.filter_sets_in[filter_set_key] = {}
+            self.filter_sets_out[filter_set_key] = {}
+
+            for j in range(1, filters_per_set + 1):
+
+                in_key = f"filter_set_{i}_in_pos_{j}"
+                in_value = builder.aOut(
+                    f"FILTER_SET:{i}:IN:{j}",
+                    initial_value=100,
+                    on_update=lambda _, i=i: self._set_in_pos(i),
+                )
+                self.filter_sets_in[filter_set_key][in_key] = in_value
+
+                out_key = f"filter_set_{i}_out_pos_{j}"
+                out_value = builder.aOut(
+                    f"FILTER_SET:{i}:OUT:{j}",
+                    initial_value=0,
+                    on_update=lambda _, i=i: self._set_out_pos(i),
+                )
+                self.filter_sets_out[filter_set_key][out_key] = out_value
+
+    async def run_forever(self) -> None:
+
+        print("Connecting to ZMQ stream...")
+
+        req_status = b'{"command":"status"}'
+
+        self._send_message(req_status)
+        # resp = await self.zmq_stream.get_response()
+
+        # if resp:
+        #     print("Connected.")
+
+        await asyncio.gather(
+            *[
+                self.monitor_responses(self.zmq_stream),
+                self.monitor_responses(self.event_stream),
```

Comment:
I don't think you can use the same method for these as you need to do different things with the messages.